.github/workflows/r-test.yml

```diff
@@ -0,0 +1,74 @@
+name: testthat
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    env:
+      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up R
+        uses: r-lib/actions/setup-r@v2
+        
+      - name: Install Ubuntu dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libcurl4-openssl-dev libharfbuzz-dev libfribidi-dev
+
+      - name: Install R dependencies
+        run: |
+          install.packages('remotes')
+          install.packages('devtools')
+          remotes::install_deps(dependencies = TRUE)
+          remotes::install_cran("rcmdcheck")
+          remotes::install_cran("httr")
+          remotes::install_github("Sage-Bionetworks/dccvalidator")
+          devtools::install()
+        shell: Rscript {0}
+
+      - name: Run tests
+        run: |
+          install.packages('testthat')
+          testthat::test_dir('tests/testthat/')
+        shell: Rscript {0}
+        continue-on-error: true
+        id: run-tests
+
+      - name: Save test results
+        if: failure()
+        run: |
+          echo "Tests failed" > test-results.log
+        id: save-results
+
+      - name: Post test results to PR
+        if: failure()
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs');
+            
+            const failedTests = fs.readFileSync('test-results.log', 'utf8');
+            const issueComment = context.payload.pull_request.number;
+      
+            if (failedTests) {
+              const body = `## :x: Test failures detected in this PR\n\n\`\`\`\n${failedTests}\n\`\`\``;
+              github.rest.issues.createComment({
+                ...context.repo,
+                issue_number: issueComment,
+                body: body
+              });
+            }
```

Comment:
This is pretty nifty!  @JessterB I imagine jira ticket to be created when actions fail right? Most of the time, I get an email when the action fails so not sure if this is 100% necessary (but it is awesome)

Reply:
I think this adds a comment to the PR in GitHub, not to a JIRA issue? 

Re. JIRA ticket creation, I poked around and found [this](https://github.com/marketplace/actions/jira-add-comment) and [this](https://github.com/atlassian/gajira), but neither is being maintained...

Reply:
Yes right now this is simply commenting on the PR itself when tests fail. I too was looking for some automatic ticket creation (i imagine its possible?) but also ran into the issue of seeing a lot of unmaintained stuff..